doc/source/container_builder.rst

```diff
@@ -0,0 +1,282 @@
+Building in a Self-Contained Environment
+========================================
+
+.. topic:: Abstract
+
+   Users building images with KIWI face problems if they want
+   to build as non root user or can't install KIWI on their
+   system due to inappropriate build host requirements or
+   project policies.
+
+   This document describes how to perform the build process in
+   a Docker container using the Dice containment build system
+   written for KIWI.
+
+   The changes to the machine to become a build host will
+   be reduced to the requirements of Dice and Docker.
+
+Requirements
```

Comment:
Although I have a some understanding about KIWI, I stumbled over some terms which I never heard of ("box/boxes"). Maybe a short terminology section could make sense here or at the end of this article?